opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java

```diff
@@ -1122,53 +1132,43 @@ private static String readOpening(InputStream in, byte[] sig)
         }
 
         int nRead = 0;
-        boolean sawNonWhitespace = false;
-        boolean lastWhitespace = false;
-        boolean postHashbang = false;
+        boolean ignoreWhiteSpace = true;
+        boolean breakOnNewLine = false;
         int r;
 
         StringBuilder opening = new StringBuilder();
         BufferedReader readr = new BufferedReader(new InputStreamReader(in, encoding), OPENING_MAX_CHARS);
+
         while ((r = readr.read()) != -1) {
-            if (++nRead > OPENING_MAX_CHARS) {
-                break;
-            }
             char c = (char) r;
-            boolean isWhitespace = Character.isWhitespace(c);
-            if (!sawNonWhitespace) {
-                if (isWhitespace) {
-                    continue;
-                }
-                sawNonWhitespace = true;
-            }
-            if (c == '\n') {
+            if (isNewLineOrMaxReadLimit(c, ++nRead, breakOnNewLine)) {
```

Comment:
it would be good to have a unit test for this method, esp, if its signature can be simplified.

Reply:
Removed this method in the latest push